test/8bit_lds_test.c

```diff
@@ -227,3 +227,78 @@ ParameterizedTest(Ld8BitSpecialTestParams *params, ld_n_hl_d8, ld_n_hl_d8, .init
     // check if PC is updated correctly
     cr_expect(eq(u8, cpu.PC, address + address_increment));
 }
+
+/******************************************************
+ *** LD HL, n                                       ***
+ ******************************************************/
+ParameterizedTestParameters(ld_hl_n, ld_hl_n) {
+    static Ld8BitTestParams params[] = {
+        {0x70, 128, 1, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, BC),
+         offsetof(DoubleWordReg, words.hi), 1},
+        {0x71, 128, 2, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, BC),
+         offsetof(DoubleWordReg, words.lo), 2},
+        {0x72, 128, 3, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, DE),
+         offsetof(DoubleWordReg, words.hi), 3},
+        {0x73, 128, 4, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, DE),
+         offsetof(DoubleWordReg, words.lo), 4},
+        {0x74, 128, 5, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, HL),
+         offsetof(DoubleWordReg, words.hi), 5},
+        {0x75, 120, 6, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, HL),
+         offsetof(DoubleWordReg, words.lo), 6},
+        {0x77, 128, 7, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), offsetof(CPU, AF),
+         offsetof(DoubleWordReg, words.hi), 7},
+    };
+
+    // generate parameter set
+    return cr_make_param_array(Ld8BitTestParams, params, sizeof(params) / sizeof(Ld8BitTestParams));
+}
+
+ParameterizedTest(Ld8BitTestParams *params, ld_hl_n, ld_hl_n, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    // setup cpu
+    uint16_t opcode_addr     = (random() % (MEM_SIZE - ROM_LIMIT)) + ROM_LIMIT;
+    uint16_t byte_write_addr = (random() % (MEM_SIZE - ROM_LIMIT)) + ROM_LIMIT;
+    uint8_t *goal_reg        = get_cpu_reg(params->l_dword_reg, params->l_word_offset);
+    uint8_t *source_reg      = get_cpu_reg(params->r_dword_reg, params->r_word_offset);
+    cpu.PC                   = opcode_addr;
+
+    mmu_write_byte(opcode_addr, params->opcode);
+    CPU_DREG_HL = byte_write_addr;
```

Comment:
Where is `goal_reg` used? 🤔  I think that `CPU_DREG_HL = byte_write_addr;` could be replaced with `goal_reg`. Then this "worker" function could be refactored to some static function with the other tests above.

Reply:
It is an artifact from the previous 8-bit loads that stuck around :D  
I initially thought about doing it in a more "general" way, but I didn't believe there were any other `LD (XY), n` calls.
But I double-checked, and there are a few others, therefore, this makes sense now

Reply:
Ah I see, very nice 👌🏻